### PR TITLE
fix comments for applyLocalityLBSetting

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
as above

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -81,11 +81,9 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, prox
 
 	clusters = append(clusters, configgen.buildOutboundClusters(env, proxy, push)...)
 
-	// compute the proxy's locality. See if we have a CDS cache for that locality.
-	// If not, compute one.
-	locality := proxy.Locality
-	if locality != nil {
-		applyLocalityLBSetting(locality, clusters, env.Mesh.LocalityLbSetting, false)
+	// apply load balancer setting fot cluster endpoints
+	if pilot.EnableLocalityLoadBalancing() && proxy.Locality != nil {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster.go]: As i added this check, should be after  #15014

**Comment by hzxuzhonghu** [comment]
not stale

File: pilot/pkg/networking/core/v1alpha3/cluster.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: the rest is fine but I don't think we want to change this behavior now (without 15014)

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: I think this should be added when the feature was introduced, it is a bug.

**Comment by howardjohn** [comment]
@hzxuzhonghu tests are failing

**Comment by hzxuzhonghu** [comment]
This should be after #15014

**Pull request merged**